src/frontends/jax/src/op/select_n.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/jax/node_context.hpp";
+#include "openvino/op/ops.hpp";
+#include "utils.hpp";
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace jax {
+namespace op {
+
+OutputVector translate_select_n(const NodeContext& context) {
+    num_inputs_check(context, 2);
+    auto num_inputs = static_cast<int>(context.get_input_size());
+    Output<Node> which = context.get_input(0);
+    if (which.get_element_type() == element::boolean) {
+        which = std::make_shared<v0::Convert>(which, element::i32);
+    }
+    OutputVector unsqueezed_cases(num_inputs-1)
+    unsqueezed_cases.reserve(num_inputs - 1);
+    for (int ind = 1; ind < num_inputs; ind++) {
+        auto case_input = context.get_input(ind);
+        auto unsqueeze = std::make_shared<v0::Unsqueeze>(
+            case_input,
+            ov::op::v0::Constant::create(element::i64, Shape{1}, std::vector<int64_t>{0})
+        );
+        unsqueezed_cases[ind-1] = unsqueeze;
+    }
+    Output<Node> cases = std::make_shared<v0::Concat>(unsqueezed_cases, 0);
+    auto shape_of = std::make_shared<v3::ShapeOf>(context.get_input(1), element::i64);
+    auto broadcast_which = std::make_shared<v1::Broadcast>(
+        which,
+        shape_of
+    );
```

Comment:
btw, if you say that all arguments must have the same shape (which and cases). may be we don't need this broadcast then? Are you sure JAX will not allow different shapes of cases and which?

Reply:
I went through the source code of select_n & also confirmed with the documentation have made sure they cant have diferent shapes & therefore reverted to previous implementation without brodcast.